src/main/resources/checkstyle-suppressions.xml

```diff
@@ -16,4 +16,13 @@
 <!--  Checkstyle Suppressions for Apache Drill -->
 <suppressions>
   <suppress files="[\\/]generated-sources[\\/]" checks="AvoidStarImport,NeedBraces"/>
+  <suppress checks="IllegalCatch"
```

Comment:
In ML this is called over-fitting: I think we're adding a bit too much complexity for too little value. I can just see myself, in a couple of months, tearing my hair out wondering why the build fails on some perfectly benign construct...

Maybe explain this rule in a comment?